app/lib/redis_configuration.rb

```diff
@@ -39,12 +39,47 @@ def namespace
   end
 
   def url
-    ENV['REDIS_URL']
+    if ENV['REDIS_SENTINEL']
+      m = ENV.fetch('REDIS_SENTINEL_MASTER', 'mymaster')
+      p = ENV.fetch('REDIS_PASSWORD', '')
+      "redis://:#{p}@#{m}"
+    else
+      ENV['REDIS_URL']
+    end
+  end
+
+  def sentinels
+    return unless ENV['REDIS_SENTINEL']
+
+    unless ENV['REDIS_SENTINEL'].include? ','
+      ips = Resolv.getaddresses(ENV['REDIS_SENTINEL'])
+      port = ENV.fetch('REDIS_SENTINEL_PORT', '26379')
+
+      ENV['REDIS_SENTINEL'] = ips.map do |ip|
+        "#{ip}:#{port}"
+      end.join(',')
+    end
+    ENV['REDIS_SENTINEL'].split(',').map do |server|
+      host, port = server.split(':')
+      { host: host, port: port.to_i }
+    end
+  end
+
+  def master_name
+    ENV.fetch('REDIS_SENTINEL_MASTER', 'mymaster')
+  end
+
+  def sentinel_mode?
+    ENV.include? 'REDIS_SENTINEL'
   end
 
   private
 
   def raw_connection
-    Redis.new(url: url, driver: :hiredis)
+    if sentinel_mode?
+      Redis.new(url: url, driver: :hiredis, sentinels: sentinels, master_name: master_name)
```

Comment:
im a bit confused why a `master_name` is given here, in the redis documentation it says that sentinel can also be a configurator, to help push new master addresses to clients once it has done a failover, so why are we giving a name here? (and why do we silently default to a dummy name `mymaster`? would that not conflict of sorts?)

Reply:
Sentinel can monitor and manage multiple masters at the same time - `mymaster` is the default name but it can be changed or more added.